cmd/manager/scap_test.go

```diff
@@ -2,105 +2,88 @@ package main
+			Expect(err).To(BeNil())
+		})
+		AfterEach(func() {
+			dataStreamFile.Close()
+		})
+
+		It("Parses content without errors", func() {
+			var err error
+			contentDS, err = parseContent(dataStreamFile)
+			Expect(err).To(BeNil())
 		})
-	}
-}
-
-func TestParseWarning(t *testing.T) {
-	debugLog = true
-	tests := map[string]struct {
-		input           string
-		expectedAPIPath string
-	}{
-		"first": {
-			input:           `<warning category="general" lang="en-US"><code class="ocp-api-endpoint">/apis/config.openshift.io/v1/oauths/cluster</code><code class="ocp-dump-location"><sub idref="xccdf_org.ssgproject.content_value_ocp_data_root" use="legacy" />/idp.yml</code>file.</warning>`,
-			expectedAPIPath: "/apis/config.openshift.io/v1/oauths/cluster",
-		},
-	}
-	for n, tc := range tests {
-		t.Run(n, func(t *testing.T) {
-			api := getPathFromWarningXML(tc.input)
-			if api != tc.expectedAPIPath {
-				t.Errorf("expected API path %s, got %s", tc.expectedAPIPath, api)
-			}
+
+		It("Gets the appropriate resource URIs", func() {
+			expected := []string{"/apis/config.openshift.io/v1/oauths/cluster"}
+			got := getResourcePaths(contentDS, "xccdf_org.ssgproject.content_profile_platform-moderate")
+			Expect(got).To(Equal(expected))
 		})
-	}
-}
-
-func TestSaveDirectoryAndFile(t *testing.T) {
-	debugLog = true
-	tests := map[string]struct {
-		root         string
-		path         string
-		expectedDir  string
-		expectedFile string
-	}{
-		"1": {
-			root:         "/tmp",
-			path:         "/apis/foo",
-			expectedDir:  "/tmp/apis",
-			expectedFile: "foo",
-		},
-		"2": {
-			root:         "/",
-			path:         "/apis/foo/bar",
-			expectedDir:  "/apis/foo",
-			expectedFile: "bar",
-		},
-		"3": {
-			root:         "/tmp/foo",
-			path:         "/apis/foo/bar/baz",
-			expectedDir:  "/tmp/foo/apis/foo/bar",
-			expectedFile: "baz",
-		},
-	}
-	for n, tc := range tests {
-		t.Run(n, func(t *testing.T) {
-			dir, file, err := getSaveDirectoryAndFileName(tc.root, tc.path)
-			if err != nil {
-				t.Error(err)
-			}
-			if dir != tc.expectedDir {
-				t.Errorf("expected dir %s, got %s", tc.expectedDir, dir)
-			}
-			if file != tc.expectedFile {
-				t.Errorf("expected file %s, got %s", tc.expectedFile, file)
-			}
+	})
+
+	Context("Parsing warnings", func() {
+		It("Gets an appropriate path from the datastream", func() {
+			expectedAPIPath := "/apis/config.openshift.io/v1/oauths/cluster"
+
+			warning := `<warning category="general" lang="en-US"><code class="ocp-api-endpoint">/apis/config.openshift.io/v1/oauths/cluster</code><code class="ocp-dump-location"><sub idref="xccdf_org.ssgproject.content_value_ocp_data_root" use="legacy" />/idp.yml</code>file.</warning>`
+			api := getPathFromWarningXML(warning)
+			Expect(api).To(Equal(expectedAPIPath))
+		})
+	})
+
+	Context("Storing files in their appropriate directories", func() {
+		It("Stores the file in the right place with the root being '/tmp'", func() {
```

Comment:
The function doesn't do any file storing, just handling the save path, so a more accurate description would be something like "Parses the save path properly.."

Reply:
Done